src/System.Diagnostics.Tracing/src/System/Diagnostics/Tracing/EventCounter.cs

```diff
@@ -92,10 +108,8 @@ private void Enqueue(float value)
                 {
                     // It is possible that two threads both think the buffer is full, but only one get to actually flush it, the other
                     // will eventually enter this code path and potentially calling Flushing on a buffer that is not full, and that's okay too.
-                    lock (_bufferedValues)
-                    {
+                    lock (this) // Lock the counter
```

Comment:
It does boil down to a contract between clients and suppliers.   The issue is 'who can call (lock obj)?'  I would are that like every other part of the contract with 'obj' the defining class is in control.  That is you are not allowed to call lock(obj), unless you know that the type of obj has said it is OK.    In this world everyone gets to do 'lock(this)' by default, and only if the class somehow documents that it is OK for users do outside the class itself.    

Your convention makes sense if the default contract is that 'lock(obj)' is allowed for everyone and basically noone gets to do lock(this) (unless the carefully document that clients can't do  lock(obj)).  

I think lock(this) is a simple useful reasonably intuitive pattern (it is the 'monitor' object pattern after all), so I hate to outlaw it, and don't think it is a hardship to say 'don't do lock(obj), unless you know the class of obj allows it (it does for object, string, arrays, which are the most common).  

But having said all this, it is not hard in this case to lock on other things (we have a string and a array that can be used.  Just say the word and I will change it.